Bug: PR #13435 breaks the pre-commit hook for any engine file

### What happened?

**PR:** [#13435 "Update eslint to v10"](https://github.com/CesiumGS/cesium/pull/13435) 

**What changed:** `no-useless-assignment` was re-enabled as `error` and `eslint-seatbelt` was added to grandfather 84 existing violations via `eslint.seatbelt.tsv` at the repo root. No explicit `seatbeltFile` was configured, so seatbelt resolves the TSV against `process.cwd()`.

**Why it breaks:** `packages/engine/lint-staged.config.js` causes lint-staged to run eslint with cwd = `packages/engine/` for engine files. eslint still finds the root config, but seatbelt looks for `packages/engine/eslint.seatbelt.tsv`, doesn't find it, treats the allow-list as empty, and the 8 grandfathered errors in `BoundingSphere.js` / `InstancingPipelineStage.js` / `Cesium3DTilesTerrainDataSpec.js` surface as hard errors blocking the commit.

**Why CI didn't catch it:** `npm run eslint` runs from the repo root where `$PWD` matches the TSV location. Only the husky pre-commit path changes cwd, and it might not have been tested.

**Impact:** Hits every contributor who stages a file under `packages/engine/`.

Workaround:

```bash
export SEATBELT_FILE="$(git rev-parse --show-toplevel)/eslint.seatbelt.tsv"
git commit
```

**Possible Fix? (Recommended by Copilot)** (`eslint.config.js`):

```js
settings: {
  seatbelt: {
    seatbeltFile: new URL("./eslint.seatbelt.tsv", import.meta.url).pathname,
  },
},
```

### Reproduction steps

1. Make a local branch
2. Edit a file in engine/
3. Stage the change
4. Try to commit it
5. Observe:

```bash
cesium2 % git commit
✔ Backed up original state in git stash (f1698443ee)
⚠ Running tasks for staged files...
❯ packages/engine/lint-staged.config.js — 102 files
❯ *.{js,cjs,mjs,ts,tsx,css,html} — 101 files
✖ eslint --cache --quiet [FAILED]
◼ prettier --write
↓ *.md — no files
✔ *.{js,cjs,mjs,ts,cts,mts} — 101 files
✔ lint-staged.config.js — 28 files
↓ Skipped because of errors from tasks.
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint --cache --quiet:

/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js
138:7 error The value assigned to 'maxSpan' is not used in subsequent statements no-useless-assignment
450:7 error The value assigned to 'maxSpan' is not used in subsequent statements no-useless-assignment
635:7 error The value assigned to 'maxSpan' is not used in subsequent statements no-useless-assignment

/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Scene/Model/InstancingPipelineStage.js
297:5 error The value assigned to 'modelMatrix' is not used in subsequent statements no-useless-assignment
303:5 error The value assigned to 'nodeComputedTransform' is not used in subsequent statements no-useless-assignment
311:5 error The value assigned to 'modelMatrix' is not used in subsequent statements no-useless-assignment
317:5 error The value assigned to 'nodeComputedTransform' is not used in subsequent statements no-useless-assignment

/Users/mark.schlosser/local/code/cesium2/packages/engine/Specs/Core/Cesium3DTilesTerrainDataSpec.js
375:3 error The value assigned to 'glbByteOffset' is not used in subsequent statements no-useless-assignment

✖ 8 problems (8 errors, 0 warnings)
```

### Sandcastle example

_No response_

### Environment

Any (assumed)


### AI acknowledgment

- [x] I used AI to generate this issue report.
- [x] (If the above is checked) I have reviewed the AI-generated content before submitting.

Try this:

```bash
unset SEATBELT_FILE
git checkout -- eslint.seatbelt.tsv
echo "// touch" >> packages/engine/Source/Core/BoundingSphere.js
git add packages/engine/Source/Core/BoundingSphere.js
git commit -m test
```

See results like this:

```bash
✔ Backed up original state in git stash (1eada76d2d)
⚠ Running tasks for staged files...
  ❯ packages/engine/lint-staged.config.js — 1 file
    ❯ *.{js,cjs,mjs,ts,tsx,css,html} — 1 file
      ✖ eslint --cache --quiet [FAILED]
      ◼ prettier --write
    ↓ *.md — no files
    ❯ *.{js,cjs,mjs,ts,cts,mts} — 1 file
      ✖ tsc [SIGKILL]
↓ Skipped because of errors from tasks.
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint --cache --quiet:

/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js
  138:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment
  450:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment
  635:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment

✖ 3 problems (3 errors, 0 warnings)


✖ Task killed: tsc
husky - pre-commit script failed (code 1)
```

Setting this env var and THEN executing the above commands will make it succeed:

```bash
export SEATBELT_FILE="$(git rev-parse --show-toplevel)/eslint.seatbelt.tsv"
```

@donmccurdy:

> Does calling ESLint directly...
> 
> ```
> npx eslint "./**/*.*js" "./**/*.*ts*" "./**/*.html"
> ```
> 
> ... without the cache flag that would otherwise be included, change the results?

It seems to also fail. See:

```bash
cesium2 % cd packages/engine
npx eslint --no-cache Source/Core/BoundingSphere.js

/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js
  138:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment
  450:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment
  635:7  error  The value assigned to 'maxSpan' is not used in subsequent statements  no-useless-assignment
```

@jjspace:

> > packages/engine/lint-staged.config.js causes lint-staged to run eslint with cwd = packages/engine/
> 
> In my understanding I don't think this statement is actually true. I could be wrong but looking at the lint-staged docs this is something you have to specifically set up to run this way and we do not.

Running the following from repo root seems to indicate cwd is packages/engine:

```bash
cesium2 % npx lint-staged --debug 2>&1 | grep -E "filepath|cwd:" | head
  cwd: undefined,
  cwd: undefined,
    filepath: '/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js',
      filepath: '/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js',
      filepath: '/Users/mark.schlosser/local/code/cesium2/packages/engine/Source/Core/BoundingSphere.js',
    cwd: '/Users/mark.schlosser/local/code/cesium2/packages/engine',
    cwd: '/Users/mark.schlosser/local/code/cesium2/packages/engine',
    cwd: '/Users/mark.schlosser/local/code/cesium2/packages/engine',
    cwd: '/Users/mark.schlosser/local/code/cesium2/packages/engine',
    cwd: '/Users/mark.schlosser/local/code/cesium2/packages/engine',
```

--- a/eslint.config.js
+++ b/eslint.config.js
@@ -5,6 +5,10 @@ import reactHooks from "eslint-plugin-react-hooks";
 import { reactRefresh } from "eslint-plugin-react-refresh";
 import tseslint from "typescript-eslint";
 import seatbelt from "eslint-seatbelt";
+import { fileURLToPath } from "node:url";
+import { dirname, join } from "node:path";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
 
 export default [
   tseslint.configs.base,
@@ -66,6 +70,11 @@ export default [
     ...configCesium.configs.browser,
     plugins: { html, "eslint-seatbelt": seatbelt },
     processor: seatbelt.processors.seatbelt,
+    settings: {
+      seatbelt: {
+        seatbeltFile: join(__dirname, "eslint.seatbelt.tsv"),
+      },
+    },
     rules: {
       ...configCesium.configs.browser.rules,
       "eslint-seatbelt/configure": "error",

